Dockerfile.deps

```diff
@@ -0,0 +1,24 @@
+# Copyright 2021 Tetrate
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
```

Comment:
maybe comment why this image? it is fairly large (>1GB) and we're also adding to it.

Reply:
I'll defer to @arkodg , I'm full on in the cargo-culting mode here.

Reply:
I guess this is needed as a base image since we trigger a `cloudbuild` build for `getenvoy` using this new  image

Reply:
added a change suggestion to save the next person looking up. thanks!